adding outlier columns (#100)
cleanlab/cleanlab-studio: docs/source/concepts/cl_columns.rst

.. _concepts_cl_columns:

Cleanlab Columns
****************

Introduction
============

Cleanlab Studio will run a number of analyses on your dataset to help identify potential data issues. Currently we look for the following issues in datasets:
- Label issues
- Outliers

> We will also continue to add new analyses soon. The raw outputs of each analysis will be available to download as Cleanlab columns by calling ``studio.download_cleanlab_columns()``.

Reviewer: ```suggestion
New analyses are being continuously added to Cleanlab Studio. The raw outputs of each analysis will be available to download as Cleanlab columns by calling ``studio.download_cleanlab_columns()``.
```

> For each analysis we will often expose two columns, one being a numeric score between 0 and 1 indicating the quality of a sample for that analysis (lower score means lower quality), and the other being a boolean, which indicates whether the sample is likely to have an issue or not. The different analyses are run separately, and hence they should be independently. For example, samples with label issues are not neccessarily outliers, and vice versa, and it is possible to have a sample with label issue marked as ``False``, but outlier marked as ``True``.

Reviewer: ```suggestion
Most analyses produce two columns, one being a numeric score between 0 and 1 indicating the quality of a sample for that analysis (lower score means lower quality), and the other being a boolean, which indicates whether the sample is likely to have an issue or not. The different analyses are run separately, and hence they should be treated independently. For example, samples with label issues are not necessarily outliers, and vice versa, and it is possible to have a sample with label issue marked as ``False``, but outlier marked as ``True``. Furthermore, it is possible to have samples with label issue marked as ``True` **and** outlier marked as ``True``: these are the raw results from the analyses, and you may interpret them as appropriate (e.g., if a data point is an outlier, it may be irrelevant that it's also likely a label issue).
```

> We give more details on the different Cleanlab columns below.

Reviewer: Presenting a summary in a rST [table](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/tables.html) would be really helpful, with details below (and perhaps hyperlinks from the column names to the details below).

Author: added table

Label Issues
============

Label quality
-------------
The ``label_quality`` column contains a score bounded between 0 and 1. The score is calculated using confident learning. The lower the score of a sample, the more likely it has a label issue.

Issue
-----

> The ``issue`` column contains a boolean value, with ``True`` indicating that the sample is likely to have a label issue. The value is obtained by thresholding the ``label_quality`` score with confident learning.

Reviewer: Rename to `label_issue`? Just `issue` will be confusing as we add more analyses (already have > 1).

This is especially important to consider for the future, when we add more analyses: should we have a consistent scheme for naming Cleanlab columns? For the boolean, doing `{name of analysis}_issue` seems good, though `outlier_issue` sounds a bit weird (just `outlier` is better). For the continuous score, using the word "quality" or "score" seems good; choosing one is a bit tricky as well. "label_quality" sounds good, "label_score" doesn't sound quite as good; however, "outlier_quality" doesn't make sense, whereas "outlier_score" is fine.

Author: renamed according to our slack discussion:
- add "is_{...}" for boolean
- add "{...}_score" for floats

The Cleanlab columns are listed below in the table. Detailed explanations of each column are provided in the following sections.

.. list-table:: 
   :widths: 50 25
   :header-rows: 1

   * - Cleanlab Column Name
     - Value Type
   * - :ref:`label_issue_score <label_issue_score>`
     - Float
   * - :ref:`is_label_issue <is_label_issue>`
     - Boolean
   * - :ref:`suggested_label <suggested_label>`
     - String
   * - :ref:`is_outlier <is_outlier>`
     - Boolean

Label Issues
============

.. _label_issue_score:

> ``label_issue_score``

Reviewer 2: should this be `label_quality_score` ? i'd expect a higher `label_issue_score` to be more likely to have a label issue
